spec/emerald/parser_spec.rb

```diff
@@ -67,16 +67,91 @@
       ast = Emerald::Parser.new(source).parse
       expect(ast).to eq([Emerald::Number.new(5)])
     end
-    
-    
+
+
     it "parses a positive float" do
       source = "+5.55"
       ast = Emerald::Parser.new(source).parse
       expect(ast).to eq([Emerald::Number.new(5.55)])
     end
   end
 
-  describe "number and atom parsing" do
+  describe "string parsing" do
+    it"parses a string ofcharacters" do
+      source = '"Hello"'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"Hello"')])
+    end
+
+    it"parses a string ofcharacters and whitespace" do
+      source = '"Hello this is a string"'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"Hello this is a string"' )])
+    end
+
+    it "parses a string of characters, whitespace and digits" do
+      source = '"H3llo this is 1 str1ng"'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"H3llo this is 1 str1ng"')])
+    end
+
+    it "parses a string of characters, whitespace, digits and symbols" do
+      source = '"H3llo, this is 1 str1ng."'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"H3llo, this is 1 str1ng."')])
+    end
+
+    it "parses unusual strings that contain a backslash immediately followed by a quotation mark" do
+      source = '"this is a weird string -> \" <- "'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"this is a weird string -> \" <- "')])
+    end
+
+    it "parses a string with a set of quotation marks are immediately after a backslash" do
+      source = '"Say \"hello\""'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"Say \"hello\""')])
+    end
+
+    it "parses a string over two lines" do
+      source = '"This is a multiline \n  string"'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"This is a multiline \n  string"')])
+    end
+
+    it "parses a multiline string" do
+      source = '"This is another
+      string across
+      three lines"'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"This is another
+      string across
+      three lines"')])
+    end
+
+    it "parses two separate strings" do
+      source = '"This is one string""This is another string"'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"This is one string"'),
+                         Emerald::String.new('"This is another string"')])
+    end
+  end
```

Comment:
Here is a test for two strings! That was a good catch :) 
I've added a couple of input combinations below as well. Would the very last one be considered a bad test - since there is an atom, number and string? Is it testing too much?